fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/PredicateSplit.java

```diff
@@ -75,33 +88,196 @@ public static PredicateSplit splitPredicate(ScalarOperator predicate) {
+                BinaryPredicateOperator predicate, RangeExtractorContext context) {
+            ScalarOperator left = predicate.getChild(0);
+            ScalarOperator right = predicate.getChild(1);
+            if (left.isColumnRef() && right.isConstantRef()) {
+                if (predicate.getBinaryType().isNotEqual()) {
+                    if (context.isAnd()) {
+                        residualPredicates.add(predicate);
+                    }
+                    return null;
+                }
+                ConstantOperator constant = (ConstantOperator) right;
+                TreeRangeSet<ConstantOperator> rangeSet = TreeRangeSet.create();
+                rangeSet.add(range(predicate.getBinaryType(), constant));
+                return Collections.singletonMap(((ColumnRefOperator) left), rangeSet);
+            } else if (left.isColumnRef() && right.isColumnRef() && context.isAnd()) {
+                if (predicate.getBinaryType().isEqual()) {
+                    columnEqualityPredicates.add(predicate);
                 } else {
-                    residualPredicates.add(scalarOperator);
+                    residualPredicates.add(predicate);
+                }
+            } else if (context.isAnd()) {
+                residualPredicates.add(predicate);
+            }
+            return null;
+        }
+
+        @Override
+        public Map<ColumnRefOperator, TreeRangeSet<ConstantOperator>> visitCompoundPredicate(
+                CompoundPredicateOperator predicate, RangeExtractorContext context) {
+            Map<ColumnRefOperator, TreeRangeSet<ConstantOperator>> columnRangeSet = new LinkedHashMap<>();
+            for (ScalarOperator child : predicate.getChildren()) {
+                boolean isAndOrigin = context.isAnd();
+                if (!predicate.isAnd()) {
+                    context.setAnd(false);
+                }
+                Map<ColumnRefOperator, TreeRangeSet<ConstantOperator>> childRange = child.accept(this, context);
+                context.setAnd(isAndOrigin);
+                if (childRange == null) {
+                    if (!context.isAnd() || !predicate.isAnd()) {
+                        return null;
+                    } else {
+                        if (!(child instanceof BinaryPredicateOperator)) {
+                            residualPredicates.add(child);
+                        }
+                        continue;
+                    }
+                }
+                if (predicate.isOr()) {
+                    // (a = 1 and b = 1) or (a = 2 and b = 2) can't convert to range
+                    if (childRange.size() > 1) {
+                        return null;
+                    }
+                    if (columnRangeSet.isEmpty()) {
+                        columnRangeSet.putAll(childRange);
+                    } else {
+                        for (ColumnRefOperator column : childRange.keySet()) {
+                            if (!columnRangeSet.containsKey(column)) {
```

Comment:
So here we can only support one column's or predicates:
MV: a = 1 or a = 2 or a = 3
Query: a = 1 or a = 2

Query: From MV where a = 1 or a = 2


Just wonder can we support this?

MV : a = 1 or b = 1 or c = 1
Query: a = 1 or b = 1

=>
Query : From MV where a = 1 or b = 1?

Reply:
In residual subsumption test allreadly support this, I will add some test case.

Reply:
mv: a > 0 OR b > 0
query: a > 5
this case not support